lib/cn/cn_tree.c

```diff
@@ -2417,104 +2425,83 @@ cn_tree_cursor_read(struct cn_cursor *cur, struct kvs_cursor_element *elem, bool
     if (HSE_UNLIKELY(cur->filter))
         key2kobj(&filter_ko, cur->filter->kcf_maxkey, cur->filter->kcf_maxklen);
 
-    nkeys_before_pfx_dbg = 0;
-
     do {
+        enum kmd_vtype vtype;
+        u32            vbidx;
+        u32            vboff;
 
         if (!bin_heap2_peek(cur->bh, (void **)&popme)) {
             *eof = (cur->eof = 1);
             return 0;
         }
 
-        /* copy out bh item before bin_heap2_pop() overwrites its
-         * element (*popme).
+        /* Copy out bh item before bin_heap2_pop() overwrites its element (*popme).
          */
         item = *popme;
-        is_tomb = item.vctx.is_ptomb;
-
         bin_heap2_pop(cur->bh, (void **)&popme);
 
         rc = cur_item_pfx_cmp(cur, &item);
-        if (rc > 0) {
-            /* [HSE_REVISIT] cursor seek should seek to max(cursor_pfx, seek_key) to avoid
-             * this code path.
-             */
-            ++nkeys_before_pfx_dbg;
-            end = true;
-            continue;
-        } else if (rc < 0) {
+        assert(rc <= 0);
+        if (rc < 0) {
             *eof = (cur->eof = 1);
             return 0;
         }
```

Comment:
I'd move the assert down here...  to after the **if (rc < 0)** block...

Reply:
Done.